project/src/components/favorite-offer-card/favorite-offer-card.tsx

```diff
@@ -1,11 +1,20 @@
 import {Offer} from '../../types/offer';
+import { getRatingInStars } from '../../utils';
+import { changeFavoriteStatus, fetchFavoriteOffersAction } from '../../store/api-actions';
+import {useDispatch} from 'react-redux';
 
 type FavoriteOfferCardProps = {
   offer: Offer;
 }
 
 function FavoriteOfferCard({offer}: FavoriteOfferCardProps): JSX.Element{
-  const {previewImage, price, title} = offer;
+  const {previewImage, price, title, rating, type, id, isFavorite} = offer;
+  const dispatch = useDispatch();
+
+  const handleFavoriteClick = () => {
+    dispatch(changeFavoriteStatus(id, Number(!isFavorite)));
+    dispatch(fetchFavoriteOffersAction());
```

Comment:
```suggestion
    dispatch(fetchFavoriteOffersAction());
```
тут ничего дополнительно перефетчивать не нужно. когда ты меняешь фейворит, у тебя с запроса возвращается обновленный оффер. и эти одним офером, тебе нужно обновить старый офер в редаксе